### PR TITLE
Header nav DR

### PR DESCRIPTION
**Pull request by peterrowland**
Fixes issue(s) #3377 .

[CircleCI](https://circleci.com/gh/18F/18f.gsa.gov/tree/BRANCH_NAME)

[:sunglasses: PREVIEW](https://federalist-c58f58dc-a215-4616-a54c-12b5eb011096.app.cloud.gov/preview/18f/18f.gsa.gov/BRANCH_NAME/)

[Preview README for this branch](https://github.com/18F/18f.gsa.gov/blob/BRANCH_NAME/README.md)

Changes proposed in this pull request:
- enabling the search bar through the uswds-jekyll template
- fix bug where blog navigation was not underlined when visiting that page
- updated navigation colors to match 18F colors
- remove site title text from header template to have logo only
- update alignment and position of nav items

Checklist:
- [x] All images being added have been optimized **_--> [learn more](https://18f.gsa.gov/styleguide/images) about how to optimize images <--_**

/cc @relevant-people

### CODE REVIEW

**Comment by lauraGgit** [comment]
@peterrowland taking a look at this, you're pulling in more than you need.

In this case, you'd will have the same visual effect if you omit both `_top_nav.scss` and `banner.html` since there we are not overriding that behavior

**Comment by peterrowland** [comment]
Looks like adding `permalink: /blog/` to blog/index.html that fixed the missing navigation link underline on the blog page breaks the blog pagination linking in other ways. Need to find a different way to fix.

**Comment by igorkorenfeld** [comment]
Very nice getting close!

A few notes
- Could we center the actual navigation part with the logo ?
Current: 
image
Target:
image

- Can you change the color of the search button to be Medium / `#046B99`/ theme-color-name. The state colors (hover, active, etc..) can follow the link pattern for colors

- I notice the line for the nav links that are hovered (or the current page) is just a little shorter than the length of the link text itself. Is that the default? I feel like it works fine on most pages, but then looks kind of odd on Blog and Guides which are shorter.

**Review by igorkorenfeld — APPROVED** [review]
Great work, the header looks good to me.

**Comment by lauraGgit** [comment]
@peterrowland looks like we're almost there! there are a number of hound flags (one of which will be ok'd if you pull back from the migrate branch and a conflict.

File: _includes/components/header.html [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        <a
+          href="{% if header.href %}{{ header.href }}{% else %}{{ site.baseurl }}/{% endif %}"
+          title="Home">
+          {% if header.logo %}
+            <img
+              class="usa-logo-img"
+              src="{% if header.logo.external %}{{ header.logo.src }}{% else %}{{ header.logo.src | relative_url }}{% endif %}"
+              alt="{{ header.logo.alt }}">
+          {% endif %}
+          <!-- <em class="usa-logo__text">
```
Comment (lauraGgit) [inline _includes/components/header.html:25]: Let's remove this and comment that this is the difference between the `uswds-jekyll` template and the one here.

File: _includes/components/header.html [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                  </button>
+                </div>
+              </form>
+            {% endif %}
+          </div>
+          {% endif %}
+
+
+        </div>
+      <!-- </div> -->
```
Comment (lauraGgit) [inline _includes/components/header.html:156]: was this our comment out or was this pulled in from the template?

File: _sass/_components/header.scss [hunk 3]
```diff
@@ -48,3 +48,37 @@
...
+
+.usa-header--basic {
+  @include at-media($theme-header-min-width) {
+    .usa-navbar {
+      position: relative;
+      width: $theme-header-logo-text-width; // TODO: review this more
+    }
+
+    .usa-nav {
+      @include u-flex("row", "align-center", "justify-end");
```
[the remark this answers is not in the document]
Comment (lauraGgit) [reply _sass/_components/header.scss:69]: @peterrowland let's update this.

File: _sass/_components/header.scss [hunk 4]
```diff
@@ -48,3 +48,37 @@
...
+      display: flex;
+      padding: 0 0 units(0.5) units(1);
+      width: 100%;
+    }
+
+    .usa-nav-container {
+      @include u-flex("align-end", "justify");
```
[the remark this answers is not in the document]
Comment (lauraGgit) [reply _sass/_components/header.scss:79]: as well as this.

File: _sass/_components/nav-buttons.scss [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    &:hover {
+      background-color: color("secondary-dark");
+      color: color("white");
+      text-decoration: none;
+    }
+  }
+}
+.usa-sidenav-list,
+.usa-footer-nav .usa-footer-primary-content {
+  .usa-button {
```
[the remark this answers is not in the document]
Comment (lauraGgit) [reply _sass/_components/nav-buttons.scss:24]: This change can be ignored because i changed the setitng in migrate.

**Review by lauraGgit — CHANGES_REQUESTED** [review]
Very minor, but would be good to clean up the hound fixes and also remove the commented code before we migration.

File: _includes/components/header.html (hunk 2 above)
Comment (peterrowland) [reply _includes/components/header.html:156]: Our comment -- There was an extra div that was forcing all the nav items into a smaller container

Comment (lauraGgit) [reply _includes/components/header.html:156]: Then let's remove it.

**Review by lauraGgit — APPROVED** [review]
thanks for making those changes 🚀

**Pull request merged**